src/modules/kubernetes.rs

```diff
@@ -97,6 +108,12 @@ fn get_aliased_name<'a>(
     }
 }
 
+#[derive(Debug)]
+enum Document {
```

Comment:
I'm not really happy that this ends up in two different codepath everywhere, just to get a bit more speed in an edge case (That's at least how I perceive this: how often do you have such big kube configs?) :-(

Personally, I would also have expected to get a common data structure from both parser.

If this is acceptable, then I would expect an explanation in the code why this was chose, so one does not have to look up the original commits to understand why two different parsers are used, either here where the data structure is defined or in the parser function.

Reply:
Sadly, the yaml-rust2 crate doesn't support serde yet and serde-yaml is unmaintained, but [there are plans](https://www.github.com/saphyr-rs/saphyr/issues/1) for this at the more actively developed branch of yaml-rust2.

Nevertheless, it might be possible to reduce duplication a bit with `macro_rules!` at least.

Reply:
I've tried to make this better by using generics to not need to duplicate the logic that finds the relevant fields for the yaml and json case and also left a comment as to why to have the two codepaths.

I do agree kubeconfigs of this size are unusual but it is not "a bit" more speed, its about 77% faster so this makes a really huge difference once you do have such a kubeconfig.